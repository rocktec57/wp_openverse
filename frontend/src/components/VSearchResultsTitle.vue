<script setup lang="ts">
import { useI18n } from "#imports"

import { computed } from "vue"

import type { SupportedMediaType, SupportedSearchType } from "~/constants/media"

import { useGetLocaleFormattedNumber } from "~/composables/use-get-locale-formatted-number"
import { getCountKey } from "~/composables/use-i18n-utilities"

const props = withDefaults(
  defineProps<{
    searchTerm: string
    searchType: SupportedSearchType
    resultCounts: [SupportedMediaType, number][]
    size?: "default" | "large"
  }>(),
  {
    size: "default",
  }
)

const getLocaleFormattedNumber = useGetLocaleFormattedNumber()
const { t } = useI18n()
const mediaLocaleCounts = computed(() =>
  props.resultCounts.reduce(
    (acc, [mediaType, count]) => {
      return {
        ...acc,
        [mediaType]: {
          count,
          countKey: getCountKey(count),
          localeCount: getLocaleFormattedNumber(count),
        },
      }
    },
    {} as Record<
      SupportedMediaType,
      { count: number; countKey: string; localeCount: string }
    >
  )
)

const _getAllMediaAriaHeading = () => {
  const imageLocaleCounts = mediaLocaleCounts.value.image
  const imageResults = t(
    `browsePage.aria.allResultsHeadingCount.image.${imageLocaleCounts.countKey}`,
    {
      localeCount: imageLocaleCounts.localeCount,
    }
  )

  const audioLocaleCounts = mediaLocaleCounts.value.audio
  const audioResults = t(
    `browsePage.aria.allResultsHeadingCount.audio.${audioLocaleCounts.countKey}`,
    {
      localeCount: audioLocaleCounts.localeCount,
    }
  )

  return t("browsePage.aria.results.all", {
    query: props.searchTerm,
    imageResults,
    audioResults,
  })
}

const ariaHeading = computed((): string => {
  switch (props.searchType) {
    case "image": {
      const { count, countKey, localeCount } = mediaLocaleCounts.value.image
      return t(`browsePage.aria.results.image.${countKey}`, {
        count,
        localeCount,
        query: props.searchTerm,
      })
    }
    case "audio": {
      const { count, countKey, localeCount } = mediaLocaleCounts.value.audio
      return t(`browsePage.aria.results.audio.${countKey}`, {
        count,
        localeCount,
        query: props.searchTerm,
      })
    }
    default:
    case "all": {
      return _getAllMediaAriaHeading()
    }
  }
})
</script>

<template>
  <h1
    class="sr-only break-words md:not-sr-only"
    :class="[
      size === 'large'
        ? 'heading-2 lg:heading-1 !leading-none lg:!leading-none'
        : 'heading-2 !leading-none',
    ]"
  >
    <span aria-hidden="true">{{ searchTerm }}</span
    ><span class="sr-only">{{ ariaHeading }}</span>
  </h1>
</template>

<style scoped>
/* Title case the user's search query */
h1:first-letter {
  text-transform: uppercase;
}
</style>
