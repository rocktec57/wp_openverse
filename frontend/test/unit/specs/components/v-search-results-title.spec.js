import { render } from "~~/test/unit/test-utils/render"

import VSearchResultsTitle from "~/components/VSearchResultsTitle.vue"

describe("VSearchResultsTitle", () => {
  let options = {
    props: {
      size: "large",
    },
    scopedSlots: {
      default: () => "zack",
    },
  }

  it("should render an h1 tag containing the correct text", async () => {
    const { findByText } = render(VSearchResultsTitle, options)
    const button = await findByText("zack")
    expect(button.tagName).toBe("H1")
  })
})
