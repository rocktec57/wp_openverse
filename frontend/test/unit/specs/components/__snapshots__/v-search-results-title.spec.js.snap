// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VSearchResultsTitle accessible heading {"searchType":"all","resultCounts":[["image",0],["audio",0]]} 1`] = `
<div>
  <h1
    class="sr-only break-words md:not-sr-only heading-2 lg:heading-1 !leading-none lg:!leading-none"
  >
    <span
      aria-hidden="true"
    >
      zack
    </span>
    <span
      class="sr-only"
    >
      All results for "zack", no images and no audio tracks.
    </span>
  </h1>
</div>
`;

exports[`VSearchResultsTitle accessible heading {"searchType":"all","resultCounts":[["image",0],["audio",10]]} 1`] = `
<div>
  <h1
    class="sr-only break-words md:not-sr-only heading-2 lg:heading-1 !leading-none lg:!leading-none"
  >
    <span
      aria-hidden="true"
    >
      zack
    </span>
    <span
      class="sr-only"
    >
      All results for "zack", no images and 10 audio tracks.
    </span>
  </h1>
</div>
`;

exports[`VSearchResultsTitle accessible heading {"searchType":"all","resultCounts":[["image",0],["audio",4300]]} 1`] = `
<div>
  <h1
    class="sr-only break-words md:not-sr-only heading-2 lg:heading-1 !leading-none lg:!leading-none"
  >
    <span
      aria-hidden="true"
    >
      zack
    </span>
    <span
      class="sr-only"
    >
      All results for "zack", no images and 4,300 audio tracks.
    </span>
  </h1>
</div>
`;

exports[`VSearchResultsTitle accessible heading {"searchType":"all","resultCounts":[["image",0],["audio",10000]]} 1`] = `
<div>
  <h1
    class="sr-only break-words md:not-sr-only heading-2 lg:heading-1 !leading-none lg:!leading-none"
  >
    <span
      aria-hidden="true"
    >
      zack
    </span>
    <span
      class="sr-only"
    >
      All results for "zack", no images and over 10,000 audio tracks.
    </span>
  </h1>
</div>
`;

exports[`VSearchResultsTitle accessible heading {"searchType":"all","resultCounts":[["image",0],["audio",10001]]} 1`] = `
<div>
  <h1
    class="sr-only break-words md:not-sr-only heading-2 lg:heading-1 !leading-none lg:!leading-none"
  >
    <span
      aria-hidden="true"
    >
      zack
    </span>
    <span
      class="sr-only"
    >
      All results for "zack", no images and over 10,001 audio tracks.
    </span>
  </h1>
</div>
`;

exports[`VSearchResultsTitle accessible heading {"searchType":"all","resultCounts":[["image",10],["audio",0]]} 1`] = `
<div>
  <h1
    class="sr-only break-words md:not-sr-only heading-2 lg:heading-1 !leading-none lg:!leading-none"
  >
    <span
      aria-hidden="true"
    >
      zack
    </span>
    <span
      class="sr-only"
    >
      All results for "zack", 10 images and no audio tracks.
    </span>
  </h1>
</div>
`;

exports[`VSearchResultsTitle accessible heading {"searchType":"all","resultCounts":[["image",10],["audio",10]]} 1`] = `
<div>
  <h1
    class="sr-only break-words md:not-sr-only heading-2 lg:heading-1 !leading-none lg:!leading-none"
  >
    <span
      aria-hidden="true"
    >
      zack
    </span>
    <span
      class="sr-only"
    >
      All results for "zack", 10 images and 10 audio tracks.
    </span>
  </h1>
</div>
`;

exports[`VSearchResultsTitle accessible heading {"searchType":"all","resultCounts":[["image",10],["audio",4300]]} 1`] = `
<div>
  <h1
    class="sr-only break-words md:not-sr-only heading-2 lg:heading-1 !leading-none lg:!leading-none"
  >
    <span
      aria-hidden="true"
    >
      zack
    </span>
    <span
      class="sr-only"
    >
      All results for "zack", 10 images and 4,300 audio tracks.
    </span>
  </h1>
</div>
`;

exports[`VSearchResultsTitle accessible heading {"searchType":"all","resultCounts":[["image",10],["audio",10000]]} 1`] = `
<div>
  <h1
    class="sr-only break-words md:not-sr-only heading-2 lg:heading-1 !leading-none lg:!leading-none"
  >
    <span
      aria-hidden="true"
    >
      zack
    </span>
    <span
      class="sr-only"
    >
      All results for "zack", 10 images and over 10,000 audio tracks.
    </span>
  </h1>
</div>
`;

exports[`VSearchResultsTitle accessible heading {"searchType":"all","resultCounts":[["image",10],["audio",10001]]} 1`] = `
<div>
  <h1
    class="sr-only break-words md:not-sr-only heading-2 lg:heading-1 !leading-none lg:!leading-none"
  >
    <span
      aria-hidden="true"
    >
      zack
    </span>
    <span
      class="sr-only"
    >
      All results for "zack", 10 images and over 10,001 audio tracks.
    </span>
  </h1>
</div>
`;

exports[`VSearchResultsTitle accessible heading {"searchType":"all","resultCounts":[["image",4300],["audio",0]]} 1`] = `
<div>
  <h1
    class="sr-only break-words md:not-sr-only heading-2 lg:heading-1 !leading-none lg:!leading-none"
  >
    <span
      aria-hidden="true"
    >
      zack
    </span>
    <span
      class="sr-only"
    >
      All results for "zack", 4,300 images and no audio tracks.
    </span>
  </h1>
</div>
`;

exports[`VSearchResultsTitle accessible heading {"searchType":"all","resultCounts":[["image",4300],["audio",10]]} 1`] = `
<div>
  <h1
    class="sr-only break-words md:not-sr-only heading-2 lg:heading-1 !leading-none lg:!leading-none"
  >
    <span
      aria-hidden="true"
    >
      zack
    </span>
    <span
      class="sr-only"
    >
      All results for "zack", 4,300 images and 10 audio tracks.
    </span>
  </h1>
</div>
`;

exports[`VSearchResultsTitle accessible heading {"searchType":"all","resultCounts":[["image",4300],["audio",4300]]} 1`] = `
<div>
  <h1
    class="sr-only break-words md:not-sr-only heading-2 lg:heading-1 !leading-none lg:!leading-none"
  >
    <span
      aria-hidden="true"
    >
      zack
    </span>
    <span
      class="sr-only"
    >
      All results for "zack", 4,300 images and 4,300 audio tracks.
    </span>
  </h1>
</div>
`;

exports[`VSearchResultsTitle accessible heading {"searchType":"all","resultCounts":[["image",4300],["audio",10000]]} 1`] = `
<div>
  <h1
    class="sr-only break-words md:not-sr-only heading-2 lg:heading-1 !leading-none lg:!leading-none"
  >
    <span
      aria-hidden="true"
    >
      zack
    </span>
    <span
      class="sr-only"
    >
      All results for "zack", 4,300 images and over 10,000 audio tracks.
    </span>
  </h1>
</div>
`;

exports[`VSearchResultsTitle accessible heading {"searchType":"all","resultCounts":[["image",4300],["audio",10001]]} 1`] = `
<div>
  <h1
    class="sr-only break-words md:not-sr-only heading-2 lg:heading-1 !leading-none lg:!leading-none"
  >
    <span
      aria-hidden="true"
    >
      zack
    </span>
    <span
      class="sr-only"
    >
      All results for "zack", 4,300 images and over 10,001 audio tracks.
    </span>
  </h1>
</div>
`;

exports[`VSearchResultsTitle accessible heading {"searchType":"all","resultCounts":[["image",10000],["audio",0]]} 1`] = `
<div>
  <h1
    class="sr-only break-words md:not-sr-only heading-2 lg:heading-1 !leading-none lg:!leading-none"
  >
    <span
      aria-hidden="true"
    >
      zack
    </span>
    <span
      class="sr-only"
    >
      All results for "zack", over 10,000 images and no audio tracks.
    </span>
  </h1>
</div>
`;

exports[`VSearchResultsTitle accessible heading {"searchType":"all","resultCounts":[["image",10000],["audio",10]]} 1`] = `
<div>
  <h1
    class="sr-only break-words md:not-sr-only heading-2 lg:heading-1 !leading-none lg:!leading-none"
  >
    <span
      aria-hidden="true"
    >
      zack
    </span>
    <span
      class="sr-only"
    >
      All results for "zack", over 10,000 images and 10 audio tracks.
    </span>
  </h1>
</div>
`;

exports[`VSearchResultsTitle accessible heading {"searchType":"all","resultCounts":[["image",10000],["audio",4300]]} 1`] = `
<div>
  <h1
    class="sr-only break-words md:not-sr-only heading-2 lg:heading-1 !leading-none lg:!leading-none"
  >
    <span
      aria-hidden="true"
    >
      zack
    </span>
    <span
      class="sr-only"
    >
      All results for "zack", over 10,000 images and 4,300 audio tracks.
    </span>
  </h1>
</div>
`;

exports[`VSearchResultsTitle accessible heading {"searchType":"all","resultCounts":[["image",10000],["audio",10000]]} 1`] = `
<div>
  <h1
    class="sr-only break-words md:not-sr-only heading-2 lg:heading-1 !leading-none lg:!leading-none"
  >
    <span
      aria-hidden="true"
    >
      zack
    </span>
    <span
      class="sr-only"
    >
      All results for "zack", over 10,000 images and over 10,000 audio tracks.
    </span>
  </h1>
</div>
`;

exports[`VSearchResultsTitle accessible heading {"searchType":"all","resultCounts":[["image",10000],["audio",10001]]} 1`] = `
<div>
  <h1
    class="sr-only break-words md:not-sr-only heading-2 lg:heading-1 !leading-none lg:!leading-none"
  >
    <span
      aria-hidden="true"
    >
      zack
    </span>
    <span
      class="sr-only"
    >
      All results for "zack", over 10,000 images and over 10,001 audio tracks.
    </span>
  </h1>
</div>
`;

exports[`VSearchResultsTitle accessible heading {"searchType":"all","resultCounts":[["image",10001],["audio",0]]} 1`] = `
<div>
  <h1
    class="sr-only break-words md:not-sr-only heading-2 lg:heading-1 !leading-none lg:!leading-none"
  >
    <span
      aria-hidden="true"
    >
      zack
    </span>
    <span
      class="sr-only"
    >
      All results for "zack", over 10,001 images and no audio tracks.
    </span>
  </h1>
</div>
`;

exports[`VSearchResultsTitle accessible heading {"searchType":"all","resultCounts":[["image",10001],["audio",10]]} 1`] = `
<div>
  <h1
    class="sr-only break-words md:not-sr-only heading-2 lg:heading-1 !leading-none lg:!leading-none"
  >
    <span
      aria-hidden="true"
    >
      zack
    </span>
    <span
      class="sr-only"
    >
      All results for "zack", over 10,001 images and 10 audio tracks.
    </span>
  </h1>
</div>
`;

exports[`VSearchResultsTitle accessible heading {"searchType":"all","resultCounts":[["image",10001],["audio",4300]]} 1`] = `
<div>
  <h1
    class="sr-only break-words md:not-sr-only heading-2 lg:heading-1 !leading-none lg:!leading-none"
  >
    <span
      aria-hidden="true"
    >
      zack
    </span>
    <span
      class="sr-only"
    >
      All results for "zack", over 10,001 images and 4,300 audio tracks.
    </span>
  </h1>
</div>
`;

exports[`VSearchResultsTitle accessible heading {"searchType":"all","resultCounts":[["image",10001],["audio",10000]]} 1`] = `
<div>
  <h1
    class="sr-only break-words md:not-sr-only heading-2 lg:heading-1 !leading-none lg:!leading-none"
  >
    <span
      aria-hidden="true"
    >
      zack
    </span>
    <span
      class="sr-only"
    >
      All results for "zack", over 10,001 images and over 10,000 audio tracks.
    </span>
  </h1>
</div>
`;

exports[`VSearchResultsTitle accessible heading {"searchType":"all","resultCounts":[["image",10001],["audio",10001]]} 1`] = `
<div>
  <h1
    class="sr-only break-words md:not-sr-only heading-2 lg:heading-1 !leading-none lg:!leading-none"
  >
    <span
      aria-hidden="true"
    >
      zack
    </span>
    <span
      class="sr-only"
    >
      All results for "zack", over 10,001 images and over 10,001 audio tracks.
    </span>
  </h1>
</div>
`;

exports[`VSearchResultsTitle accessible heading {"searchType":"audio","resultCounts":[["audio",0],["image",0]]} 1`] = `
<div>
  <h1
    class="sr-only break-words md:not-sr-only heading-2 lg:heading-1 !leading-none lg:!leading-none"
  >
    <span
      aria-hidden="true"
    >
      zack
    </span>
    <span
      class="sr-only"
    >
      No audio tracks for "zack"
    </span>
  </h1>
</div>
`;

exports[`VSearchResultsTitle accessible heading {"searchType":"audio","resultCounts":[["audio",10],["image",0]]} 1`] = `
<div>
  <h1
    class="sr-only break-words md:not-sr-only heading-2 lg:heading-1 !leading-none lg:!leading-none"
  >
    <span
      aria-hidden="true"
    >
      zack
    </span>
    <span
      class="sr-only"
    >
      10 audio tracks for "zack".
    </span>
  </h1>
</div>
`;

exports[`VSearchResultsTitle accessible heading {"searchType":"audio","resultCounts":[["audio",4300],["image",0]]} 1`] = `
<div>
  <h1
    class="sr-only break-words md:not-sr-only heading-2 lg:heading-1 !leading-none lg:!leading-none"
  >
    <span
      aria-hidden="true"
    >
      zack
    </span>
    <span
      class="sr-only"
    >
      4,300 audio tracks for "zack".
    </span>
  </h1>
</div>
`;

exports[`VSearchResultsTitle accessible heading {"searchType":"audio","resultCounts":[["audio",10000],["image",0]]} 1`] = `
<div>
  <h1
    class="sr-only break-words md:not-sr-only heading-2 lg:heading-1 !leading-none lg:!leading-none"
  >
    <span
      aria-hidden="true"
    >
      zack
    </span>
    <span
      class="sr-only"
    >
      Over 10,000 audio tracks for "zack".
    </span>
  </h1>
</div>
`;

exports[`VSearchResultsTitle accessible heading {"searchType":"audio","resultCounts":[["audio",10001],["image",0]]} 1`] = `
<div>
  <h1
    class="sr-only break-words md:not-sr-only heading-2 lg:heading-1 !leading-none lg:!leading-none"
  >
    <span
      aria-hidden="true"
    >
      zack
    </span>
    <span
      class="sr-only"
    >
      Over 10,001 audio tracks for "zack".
    </span>
  </h1>
</div>
`;

exports[`VSearchResultsTitle accessible heading {"searchType":"image","resultCounts":[["image",0],["audio",0]]} 1`] = `
<div>
  <h1
    class="sr-only break-words md:not-sr-only heading-2 lg:heading-1 !leading-none lg:!leading-none"
  >
    <span
      aria-hidden="true"
    >
      zack
    </span>
    <span
      class="sr-only"
    >
      No image results for "zack"
    </span>
  </h1>
</div>
`;

exports[`VSearchResultsTitle accessible heading {"searchType":"image","resultCounts":[["image",10],["audio",0]]} 1`] = `
<div>
  <h1
    class="sr-only break-words md:not-sr-only heading-2 lg:heading-1 !leading-none lg:!leading-none"
  >
    <span
      aria-hidden="true"
    >
      zack
    </span>
    <span
      class="sr-only"
    >
      10 image results for "zack".
    </span>
  </h1>
</div>
`;

exports[`VSearchResultsTitle accessible heading {"searchType":"image","resultCounts":[["image",4300],["audio",0]]} 1`] = `
<div>
  <h1
    class="sr-only break-words md:not-sr-only heading-2 lg:heading-1 !leading-none lg:!leading-none"
  >
    <span
      aria-hidden="true"
    >
      zack
    </span>
    <span
      class="sr-only"
    >
      4,300 image results for "zack".
    </span>
  </h1>
</div>
`;

exports[`VSearchResultsTitle accessible heading {"searchType":"image","resultCounts":[["image",10000],["audio",0]]} 1`] = `
<div>
  <h1
    class="sr-only break-words md:not-sr-only heading-2 lg:heading-1 !leading-none lg:!leading-none"
  >
    <span
      aria-hidden="true"
    >
      zack
    </span>
    <span
      class="sr-only"
    >
      Over 10,000 image results for "zack".
    </span>
  </h1>
</div>
`;

exports[`VSearchResultsTitle accessible heading {"searchType":"image","resultCounts":[["image",10001],["audio",0]]} 1`] = `
<div>
  <h1
    class="sr-only break-words md:not-sr-only heading-2 lg:heading-1 !leading-none lg:!leading-none"
  >
    <span
      aria-hidden="true"
    >
      zack
    </span>
    <span
      class="sr-only"
    >
      Over 10,001 image results for "zack".
    </span>
  </h1>
</div>
`;

exports[`VSearchResultsTitle should render an h1 tag containing the correct text 1`] = `
<div>
  <h1
    class="sr-only break-words md:not-sr-only heading-2 lg:heading-1 !leading-none lg:!leading-none"
  >
    <span
      aria-hidden="true"
    >
      zack
    </span>
    <span
      class="sr-only"
    >
      All results for "zack", 10 images and 10 audio tracks.
    </span>
  </h1>
</div>
`;
